classes_inheritance_and_modules/task_1.rb

```diff
@@ -0,0 +1,124 @@
+#1. Go through methods lesson and demos once again.
+# Then create 5 simple classes structure with 3 inheritance levels
+# and write unit tests which demonstrate
+# different methods availability levels (public, private, protected)
+
+#=============#
+#lvl 1
+class Animal
+  attr_accessor :animal_name, :animal_class
+
+  def what_eats(animal)
+    puts "#{animal.animal_class.capitalize + 's'} eat #{animal.food_type}"
+    what_eat = animal.food_type
+    what_eat
+  end
+
+  def eat
+    food = food_type
+    puts "#{animal_name.capitalize} is eating #{food} now"
+    food
+  end
+
+  def sleep
+    puts "#{animal_name.capitalize} sleeps"
+    is_sleeping = true
+    is_sleeping
+  end
+
+# protected method
+  protected
+  def food_type
+    case animal_class
+      when 'bird'
+        food = 'corn'
+      when 'fish'
+        food = 'water plants'
+      when 'reptile'
+        food = 'other animals'
+      else
+        raise 'This is unknown animal'
+    end
+    food
+  end
+end
+
+#=============#
+#lvl 2
+class Fish < Animal
+  def initialize
+    @animal_class = 'fish'
+  end
+
+  def swim
+    puts "#{animal_name.capitalize} swims"
+    is_swim = true
+    is_swim
+  end
+end
+
+
+class Tetrapod < Animal
+  def night_sleeping
+    going_sleep
+    3.times do
+      sleep
+    end
+    wake_up
+    is_slept_already = true
```

Comment:
if fact this line and the next one are redundant
each method in ruby will always return result of the last method line
so if we have this kind of method:

```
def my_method
  do_something
  true
end
```

is the same as


``` 
def my_method
  do_something
  my_var = true
  my_var
end
```